components/MultiPanel/index.ts

```diff
@@ -1,12 +1,183 @@
 import './styles.css';
 
+/**
+ * A multi-panel view that the user can add any number of 'panels'. 
+ * 'a panel' consists of two elements. even index element begomes heading, 
+ * and odd index element becomes the expandable content.
+ */
 export default class MultiPanel extends HTMLElement {
+
   constructor () {
     super();
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
   }
 
   connectedCallback () {
+    this._childrenChange();
+
+    const children : Element[] = Array.from(this.children);
+    for (let i : number = 0; i < children.length; i++){
+      const child : Element = children[i]
+
+      if(i % 2 === 0){  
+        // even index = heading element
+        child.classList.add('panel-heading');
+        
+        // for A11y
+        child.id = `panel-heading-${i}`;
+        child.setAttribute('tabindex', '-1');
+        child.setAttribute('aria-controls', `panel-content-${i}`);
+      } else {
+        // odd index = content element
+        child.classList.add('panel-content');
+
+        // for A11y
+        child.id = `panel-content-${i-1}`;
+        child.setAttribute('aria-labelledby', `panel-heading-${i-1}`);
+      } 
+    }
+
+    // make the first heading focusable.
+    children[0].setAttribute('tabindex', '0');
+
+    // add EventListners
+    this.addEventListener('click', this._onClick);
+    this.addEventListener('keydown', this._onKeyDown);
+  }
+
+  // remove EventListeners
+  disconnectedCallback() {
+    this.removeEventListener('click', this._onClick);
+    this.removeEventListener('keydown', this._onKeyDown);
   }
+  
+  // Click event handler
+  _onClick(event) {
+    const heading : Element = event.target;
+    if (this._isHeading(heading)) {
+      this._expand(heading);
+    }
```

Comment:
This breaks if the user clicks an element inside the heading. As in:

```html
<h1>Hello <em>world</em></h1>
```

Clicking `world` won't work.

Instead of `_isHeading`, maybe write something like `_getClosestHeading`? `el.closest('multi-panel > *')` will help here.

`_getClosestHeading` should return undefined/null if the user didn't click within a heading.